README.md

```diff
@@ -207,23 +220,293 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
+
+* `generateRWrappers`
+* `generateRdFiles`
+
+The .Rd file generation must happen at the time the package is built, not at the time the package is loaded. The R function and constructor wrappers must happen at the time the package is loaded. These functions should be called with the similar parameters to ensure that all R wrapped functions have corresponding reference documentation.
+
+## Examples:
+
+In this example, I will demonstrate how we generate [synapser](https://github.com/Sage-Bionetworks/synapser) and [synapserutils](https://github.com/Sage-Bionetworks/synapserutils) packages by wrapping Python package [synapsePythonClient](https://github.com/Sage-Bionetworks/synapsePythonClient).
+
+The `synapseutils` module in `synapsePythonClient` package has the following structure:
+```
+    module: synapseutils
+        module: copy
+            function: copy
+            function: copyWiki
+            function: copyFileHandles
+        module: sync
+            function: syncToSynapse
+            function: syncFromSynapse
+        module: monitor
+            function: notifyMe
+```
+
+### Expose all functions and classes within a Python module
+
+#### Generate .Rd files
+
+To generate the R package `synapserutils`, our first attempt is to exposing all functions under `synapseutils` Python module. 
+
+In the `synapserutils` package, we add a `configure` file with the following content:
+```
+#!/bin/sh
+
+export PWD_FROM_R=${ALT_PWD-`pwd`}
+
+# build the .Rd files
+Rscript --vanilla path/to/createRdFiles.R $PWD_FROM_R
+```
+
+Then in `createRdFiles.R`, we calls `generateRdFiles` as follows:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseutils",
+                module = "synapseutils")
+```
+
+Where:
+* `srcRootDir` is the path to `synapserutils` directory. The directory must exist prior to this call. `generateRdFiles` will create a folder `auto-man` and write the generated .Rd files in this folder.
+* `pyPkg` is the name of the Python package that needs to be imported, and
+* `module` is the name of the Python module to be wrapped. This parameter can take the same value as `pyPkg` or a module within the Python package. The value that is passed to `module` must be a fully qualified name.
+
+#### Generate R wrappers
+
+To generate R wrappers for Python functions and constructors, we need to add an `.onLoad` hook in the R package. This hook is defined in `synapserutils/R/zzz.R` file:
+```r
+# For the R wrappers to be available in `synapserutils` package namespace,
+# `setGeneric` must be defined in the `synapserutils` package. Therefore,
+# we need to define it in `synapserutils` and pass it through `generateRWrappers`.
+callback <- function(name, def) {
+  setGeneric(name, def)
+}
+
+.onLoad <- function(libname, pkgname) {
+  generateRWrappers(pyPkg = "synapseutils",
+                    module = "synapseutils",
+                    setGenericCallback = callback)
+}
+```
```

Comment:
The values for the pyPkg and module parameters to generateRWrappers must match those passed to generateRdFiles() so that the reference doc's match the generated R functions.